vms/platformvm/txs/executor/proposal_tx_preference.go

```diff
@@ -0,0 +1,188 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package executor
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/snow/uptime"
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/vms/platformvm/reward"
+	"github.com/ava-labs/avalanchego/vms/platformvm/state"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+)
+
+var (
+	_ txs.Visitor = (*ProposalTxPreference)(nil)
+
+	ErrMissingStakerTx                      = errors.New("failed to get staker transaction")
+	ErrUnexpectedStakerTransactionType      = errors.New("unexpected staker transaction type")
+	ErrStakerWithoutPrimaryNetworkValidator = errors.New("failed to get primary network validator of staker")
+	ErrMissingSubnetTransformation          = errors.New("failed to get subnet transformation")
+	ErrCalculatingUptime                    = errors.New("failed calculating uptime")
+)
+
+type ProposalTxPreference struct {
```

Comment:
I'm a bit torn on if we really need a visitor here. I don't think it's likely we'll add any other oracle txs... So we could probably get away with a simple type check.